server/src/test/java/org/opensearch/index/store/remote/file/OnDemandBlockSnapshotIndexInputTests.java

```diff
@@ -0,0 +1,510 @@
+        TestGroup.testReadVInt(blockedSnapshotFile, blockSize, fileSize);
+        TestGroup.testSeek(blockedSnapshotFile, blockSize, fileSize);
+        TestGroup.testReadByteWithPos(blockedSnapshotFile, blockSize, fileSize);
+        TestGroup.testReadShortWithPos(blockedSnapshotFile, blockSize, fileSize);
+        TestGroup.testReadIntWithPos(blockedSnapshotFile, blockSize, fileSize);
+        TestGroup.testReadLongWithPos(blockedSnapshotFile, blockSize, fileSize);
+        TestGroup.testReadBytes(blockedSnapshotFile, blockSize, fileSize);
+    }
+
+    // create OnDemandBlockSnapshotIndexInput for each block size
+    private static OnDemandBlockSnapshotIndexInput createOnDemandBlockSnapshotIndexInput(int blockSizeShift, long fileSize) {
+
+        // file info should be initialized per test method since file size need to be calculated
+        fileInfo = mock(BlobStoreIndexShardSnapshot.FileInfo.class);
+        byteSizeValue = mock(ByteSizeValue.class);
+
+        int blockSize = 1 << blockSizeShift;
+
+        when(fileInfo.partSize()).thenReturn(byteSizeValue);
+        when(fileInfo.physicalName()).thenReturn(FILE_NAME);
+        // part size is related to snapshot, useless here
+        when(byteSizeValue.getBytes()).thenReturn(1073741824L);
+        when(fileInfo.length()).thenReturn(fileSize);
+        doAnswer(invocation -> {
+            BlobFetchRequest blobFetchRequest = invocation.getArgument(0);
+            return CompletableFuture.completedFuture(
+                blobFetchRequest.getDirectory().openInput(blobFetchRequest.getFileName(), IOContext.READ)
+            );
+        }).when(transferManager).asyncFetchBlob(any());
+
+        FSDirectory directory = null;
+        try {
+            cleanDirectory(path);
+            // use MMapDirectory to create block
+            directory = new MMapDirectory(path, lockFactory);
+        } catch (IOException e) {
+            fail("fail to create MMapDirectory: " + e.getMessage());
+        }
+
+        initBlockFiles(blockSize, fileSize, directory);
+
+        return new OnDemandBlockSnapshotIndexInput(
+            OnDemandBlockIndexInput.builder()
+                .resourceDescription(RESOURCE_DESCRIPTION)
+                .offset(BLOCK_SNAPSHOT_FILE_OFFSET)
+                .length(fileSize)
+                .blockSizeShift(blockSizeShift)
+                .isClone(IS_CLONE),
+            fileInfo,
+            directory,
+            transferManager
+        );
+    }
+
+    private static void initBlockFiles(int blockSize, long fileSize, FSDirectory fsDirectory) {
+        // for this test, it's safe to convert long to int
+        int intFileSize = (int) fileSize;
+        int numOfBlocks = intFileSize / blockSize;
+
+        int sizeOfLastBlock = intFileSize % blockSize;
+
+        try {
+
+            // block size will always be an integer multiple of frame size
+            // write 48, 49 alternatively
```

Comment:
Is this correct? It looks like this writes 48, -80 alternatively.

Reply:
yeah corrected